# Identify spatial domains based on graph attention autoencoder and contrastive learning in spatial transcriptomics data
![GCLMVST](https://github.com/XiuYuhan/GCLMVST/blob/main/GCLMVST-main/GCLMVST-model.jpg)
# Overview
GCLMVST is a novel multi-view single-cell spatial transcriptomics data analysis framework that integrates graph attention autoencoders with contrastive learning, aiming to enhance the precision and robustness of spatial structure analysis by accurately modeling cellular expression patterns and spatial dependencies in Spatial Transcriptomics (ST) data. This framework combines Edge-Weighted Graph Attention Mechanism (EW-GAT) with graph contrastive learning, dynamically adjusting edge weights in the graph to adaptively capture the synergistic expression of spatially proximal cells, while also identifying cell-to-cell cooperative expression across distant spatial regions. In GCLMVST, the graph attention autoencoder effectively encodes spatial information from multiple perspectives, enabling the integration of expression features across various spatial scales. The graph contrastive learning component further strengthens the model's ability to discern spatial structures by constructing positive and negative view pairs for contrast, facilitating contrastive learning between spatial domains. By utilizing this framework, GCLMVST is capable of handling complex spatial tissue architectures from multiple views, significantly improving single-cell clustering accuracy in spatial transcriptomics and revealing spatial dependencies and heterogeneity among cells. GCLMVST not only efficiently handles spatial data at varying resolutions but also detects subtle spatial structure variations within complex tissues. It possesses powerful data integration capabilities, capable of correcting batch effects and adapting to cross-sample spatial transcriptomics analyses. This makes it highly applicable to disease diagnosis, tissue subtyping, and precision medicine, providing deeper spatial transcriptomic insights into complex tissues and disease mechanisms.

# Data
All datasets are open access. The DLPFC datasets are available online at http://research.libd.org/spatialLIBD/.
<br>Mouse olfactory bulb of Stereo-seq dataset is available online at https://github.com/JinmiaoChenLab/SEDR_analyses/.
<br>Adult Mouse Brain of 10X Visium dataset is available online at https://support.10xgenomics.com/spatial-gene-expression/datasets.
<br>Human breast cancer of 10X Visium dataset is available online at https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/.
